components/camel-iec60870/src/main/java/org/apache/camel/component/iec60870/client/ClientConnection.java

```diff
@@ -99,7 +94,19 @@ public ClientConnection(final String host, final int port, final ClientOptions o
     public void start() {
         final DataModule dataModule = new DataModule(this.dataHandler, this.options.getDataModuleOptions());
         final ModulesFactory factory = () -> Arrays.asList(dataModule, new DiscardAckModule());
-        this.client = new AutoConnectClient(this.host, this.port, this.options.getProtocolOptions(), factory, this.stateListener);
+        final CountDownLatch latch = new CountDownLatch(1);
+        this.client = new AutoConnectClient(this.host, this.port, this.options.getProtocolOptions(), factory, new StateListener() {
+            @Override
+            public void stateChanged(final State state, final Throwable e) {
+                if (state == State.CONNECTED)
+                    latch.countDown();
+            }
+        });
+        try {
+            latch.await(2000, TimeUnit.MILLISECONDS);
```

Comment:
Is 2 seconds not a bit low? Why not have 5 or 10 seconds, and maybe have it configurable as an option on the endpoint/component (or both places)

Reply:
The original impl fails with NPE when the client does not connect async in time before the first message comes in. I think 2secs is good enough for now (i.e. for fuse-7.6) on the assumption that the server is reachable at all. 

Making this more robust such that it can deal with a connection becoming available late, being unreliable, and connectivity other connectivity issues during the lifetime of the endpoint is another (jira) issue I think

Reply:
Yeah sure to make it more robust may be another JIRA, but you can use Camels error handler to try the producer again to mitigate some connectivity issues etc. 

The 2 sec is hardcoded and any hardcoded values like that is always bad. Can you make it configurable as requested.